science/bali-phy/Portfile

```diff
@@ -1,110 +1,41 @@
 
-configure.args      --with-system-boost --enable-cairo
-configure.env-append BOOST_SUFFIX=-mt
+#build: ${build.cmd} ${build.target} ${build.args} = ninja -C builddir
+build.cmd	    ninja
+build.target	    
+build.args          -C builddir
 
-variant no_cairo description {disable drawing routines} {
-    depends_lib-delete      path:lib/pkgconfig/cairo.pc:cairo
-    configure.args-delete   --enable-cairo
-    patchfiles-delete       src-tools-draw-tree.patch
-}
-
-variant openmpi description {enable OpenMPI parallelization} {
-    depends_lib-append      port:openmpi
-    configure.args-append   --with-mpi
-}
-
-set gcc_versions {4.3 4.4 4.5 4.6 4.7 4.8 4.9}
-set default_fortran_variant +gcc48
-
-foreach ver ${gcc_versions} {
-    set ver_no_dot [join [split ${ver} "."] ""]
-    set conflicts_list {}
-    foreach over ${gcc_versions} {
-        if {${ver} eq ${over}} {
-            continue
-        }
-
-        set over_no_dot [join [split ${over} "."] ""]
-        lappend conflicts_list gcc${over_no_dot}
-    }
-
-    variant gcc${ver_no_dot} description "build with gfortran from gcc${ver_no_dot}" \
-        conflicts {*}${conflicts_list} {}
-
-    if {[variant_isset gcc${ver_no_dot}]} {
-        if {${default_fortran_variant} != "+gcc${ver_no_dot}"} {
-            set default_fortran_variant ""
-        }
-    }
-}
-
-if {${default_fortran_variant} != ""} {
-    default_variants-append "${default_fortran_variant}"
-}
-
-foreach ver ${gcc_versions} {
-    set ver_no_dot [join [split ${ver} "."] ""]
-
-    if {[variant_isset gcc${ver_no_dot}]} {
-        depends_lib-append path:lib/libgcc/libgcc_s.1.dylib:libgcc
-        depends_build-append port:gcc${ver_no_dot}
-
-        configure.fc  ${prefix}/bin/gfortran-mp-${ver}
-        configure.f77 ${prefix}/bin/gfortran-mp-${ver}
-        configure.f90 ${prefix}/bin/gfortran-mp-${ver}
-    }
-}
-
-# The project doesn't even configure with clang.  Based on how long it's
-# been broken, I doubt this project is even used much, so it may not be
-# worh the effort to fix
-#
-# conftest.cpp:60:11: error: no type named 'default_name_check' in 'boost::filesystem::path'
-# fs::path::default_name_check(fs::portable_posix_name);
-# ~~~~~~~~~~^
-# conftest.cpp:60:34: error: definition or redeclaration of 'portable_posix_name' not allowed inside a function
-# fs::path::default_name_check(fs::portable_posix_name);
-#                              ~~~~^
-# 2 errors generated.
-compiler.blacklist *clang*
-
-platform darwin {
-    if {${os.major} >= 13} {
-        # TODO: Test Mavericks once the clang build failures are addressed
-
-        depends_build
-        depends_lib
-        pre-fetch {
-            ui_error "$name does not build on Mavericks or later."
-            error "unsupported platform"
-        }
-    }
-}
-
-livecheck.type  regex
-livecheck.url   ${master_sites}
-livecheck.regex {BAli-Phy (\d+\.\d+\.\d+) released}
```

Comment:
Given the amount of code being removed, you should at least provide some explanation in the commit message. At least mention that building with `clang` has finally been possible etc.

What's with OpenMPI, cairo, Fortran?

Reply:
Fortran is irrelevant - the project never used fortran.  I'm not sure why it was originally there.
OpenMPI is more of an experimental build option - I don't think providing an MPI variant is useful.
Cairo is now autodetected, so you don't need to pass a configure flag to enable it.  I also eliminated the flag.  Is there value in providing a no-cairo variant?
I provide this in the description of the next pull request.

Reply:
If you are author of the software, you need to know whether the version without cairo is even useful. Cairo is needed so often in other ports that I wouldn't bother creating a separate variant for the sake of theoretically saving a few bytes. We sometimes exclude super heavy dependencies which are not really critical for functionality of software.

Using `no_cairo` was wrong anyway as we no longer use negative variants.

What do you mean with the next pull request? Don't open another pull request, do the fixes in this one.

Reply:
I saw what you wrote about not opening new pull requests.  I mostly use git from the command line so I'm not familiar with github's web interface.  I presume that you can suggest a second, alternative patch but in the same pull request?  I have not figured out how to do this yet.

Reply:
Just push the changes to the same branch ([patch3](https://github.com/bredelings/macports-ports/tree/patch-3) of your clone). You can either add new commits or use `git commit --amend` and then force push to `patch-3`.